arviz/tests/test_data.py

```diff
@@ -192,6 +192,58 @@ def test_concat(copy, inplace, sequence):
         assert id(new_idata.observed_data) == id(idata3.observed_data)
 
 
+@pytest.mark.parametrize("dim", ["chain", "draw"])
+@pytest.mark.parametrize("copy", [True, False])
+@pytest.mark.parametrize("inplace", [True, False])
+@pytest.mark.parametrize("sequence", [True, False])
+@pytest.mark.parametrize("reset_dim", [True, False])
+def test_concat_dim(dim, copy, inplace, sequence, reset_dim):
+    idata1 = from_dict(
+        posterior={"A": np.random.randn(2, 10, 2), "B": np.random.randn(2, 10, 5, 2)},
+        observed_data={"C": np.random.randn(100), "D": np.random.randn(2, 100)},
+    )
+    if inplace:
+        original_idata1_id = id(idata1)
+    idata2 = from_dict(
+        posterior={"A": np.random.randn(2, 10, 2), "B": np.random.randn(2, 10, 5, 2)},
+        observed_data={"C": np.random.randn(100), "D": np.random.randn(2, 100)},
+    )
+    idata3 = from_dict(
+        posterior={"A": np.random.randn(2, 10, 2), "B": np.random.randn(2, 10, 5, 2)},
+        observed_data={"C": np.random.randn(100), "D": np.random.randn(2, 100)},
+    )
+    # basic case
+    assert (
+        concat(idata1, idata2, dim=dim, copy=copy, inplace=False, reset_dim=reset_dim) is not None
+    )
+    if sequence:
+        new_idata = concat(
+            (idata1, idata2, idata3), copy=copy, dim=dim, inplace=inplace, reset_dim=reset_dim
+        )
+    else:
+        new_idata = concat(
+            idata1, idata2, idata3, dim=dim, copy=copy, inplace=inplace, reset_dim=reset_dim
+        )
+    if inplace:
+        assert new_idata is None
+        new_idata = idata1
+    assert new_idata is not None
+    test_dict = {"posterior": ["A", "B"], "observed_data": ["C", "D"]}
+    fails = check_multiple_attrs(test_dict, new_idata)
+    assert not fails
+    if inplace:
+        assert id(new_idata) == original_idata1_id
+    else:
+        assert id(new_idata) != id(idata1)
+    print(getattr(new_idata.posterior, dim).size)

```

Comment:
Not the biggest of deals. Can this print be changed to a log statement?

Reply:
There should be no print, it's my debug line